smart-contract/assembly/contracts/deweb-interface.ts

```diff
@@ -0,0 +1,129 @@
+import { balance, Context, transferCoins } from '@massalabs/massa-as-sdk';
+import {
+  _onlyOwner,
+  _setOwner,
+} from '@massalabs/sc-standards/assembly/contracts/utils/ownership-internal';
+import { ChunkPost, ChunkGet, PreStore } from './serializable/Chunk';
+import { Args, u32ToBytes } from '@massalabs/as-types';
+import {
+  _getFileChunk,
+  _getTotalChunk,
+  _removeChunksRange,
+  _removeFile,
+  _setFileChunk,
+  _setTotalChunk,
+} from './internals/chunks';
+import { FILES_PATH_LIST } from './internals/const';
+import { _pushFilePath } from './internals/file-list';
+
+/**
+ * Initializes the smart contract.
+ * Sets the contract deployer as the owner and initializes an empty file path list.
+ * @param _ - Unused parameter (required).
+ */
+export function constructor(_: StaticArray<u8>): void {
+  if (!Context.isDeployingContract()) return;
+  _setOwner(Context.caller().toString());
+  FILES_PATH_LIST.set([]);
+}
+
+/**
+ * Stores file chunks in the contract storage.
+ * Only the contract owner can call this function.
+ * @param _binaryArgs - Serialized arguments containing an array of ChunkPost objects.
+ * @throws If the chunks are invalid or if the caller is not the owner.
+ */
+export function storeFileChunks(_binaryArgs: StaticArray<u8>): void {
+  _onlyOwner();
+  const args = new Args(_binaryArgs);
+  const chunks = args
+    .nextSerializableObjectArray<ChunkPost>()
+    .expect('Invalid chunks');
+
+  for (let i = 0; i < chunks.length; i++) {
+    _setFileChunk(chunks[i].filePath, chunks[i].index, chunks[i].data);
+  }
+}
+
+export function preStoreFileChunks(_binaryArgs: StaticArray<u8>): void {
```

Comment:
Missing doc here, especially important since this function does particular stuff